src/app/view/components/_common/dot-wizard/dot-wizard.component.ts

```diff
@@ -0,0 +1,202 @@
+import {
+    Component,
+    ComponentFactoryResolver,
+    Input,
+    OnInit,
+    QueryList,
+    ViewChild,
+    ViewChildren
+} from '@angular/core';
+import { DotContainerReferenceDirective } from '@directives/dot-container-reference/dot-container-reference.directive';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { ComponentRef } from '@angular/core/src/linker/component_factory';
+import { Dialog } from 'primeng/dialog';
+import { DotWizardStep } from '@models/dot-wizard-step/dot-wizard-step.model';
+import { DotWizardService } from '@services/dot-wizard/dot-wizard.service';
+import { OnDestroy } from '@angular/core/src/metadata/lifecycle_hooks';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'dot-wizard',
+    templateUrl: './dot-wizard.component.html',
+    styleUrls: ['./dot-wizard.component.scss']
+})
+export class DotWizardComponent implements OnInit, OnDestroy {
+    wizardData: { [key: string]: string };
+    dialogActions: DotDialogActions;
+    transform = '';
+
+    @Input() steps: DotWizardStep[] = [];
+    @ViewChildren(DotContainerReferenceDirective)
+    formHosts: QueryList<DotContainerReferenceDirective>;
+    @ViewChild('dialog') dialog: Dialog;
+
+    private currentStep = 0;
+    private componentsHost: DotContainerReferenceDirective[];
+    private stepsValidation: boolean[];
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private componentFactoryResolver: ComponentFactoryResolver,
+        private dotMessageService: DotMessageService,
+        private dotWizardService: DotWizardService
+    ) {}
+
+    ngOnInit() {
+        this.dotWizardService.showDialog$.subscribe(data => {
+            this.steps = data;
+            setTimeout(() => {
+                this.loadComponents();
```

Comment:
What if you this in the `afterViewInit` or something like that?

Reply:
if gives `ngAfterViewChecked` error in the components that we try to load.